src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CISShadowMonitor.java

```diff
@@ -217,56 +225,74 @@ private void processCISShadow(ShadowDeltaUpdatedEvent event) {
     }
 
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    private synchronized void processCISShadow(int version, Map<String, Object> desiredState) {
-        if (version == lastVersion) {
-            LOGGER.atInfo().kv(VERSION, version).log("Already processed version. Skipping cert re-generation");
-            updateCISShadowReportedState(version, desiredState);
-            return;
-        }
-
-        LOGGER.atInfo().log("New CIS version: {}", version);
-
+    private void processCISShadow(int version, Map<String, Object> desiredState) {
         // NOTE: This method executes in an MQTT CRT thread. Since certificate generation is a compute intensive
-        // operation (particularly on low end devices) it is imperative that we process this event asynchronously
+        // operation (particularly on low-end devices) it is imperative that we process this event asynchronously
         // to avoid blocking other MQTT subscribers in the Nucleus
         CompletableFuture.runAsync(() -> {
-            try {
-                RetryUtils.runWithRetry(GET_CONNECTIVITY_RETRY_CONFIG, connectivityInfoProvider::getConnectivityInfo,
-                        "get-connectivity", LOGGER);
-            } catch (InterruptedException e) {
-                LOGGER.atDebug().kv(VERSION, version).cause(e)
-                        .log("Retry workflow for getting connectivity info interrupted");
-                Thread.currentThread().interrupt();
-                return;
-            } catch (Exception e) {
-                LOGGER.atError().kv(VERSION, version).cause(e)
-                        .log("Failed to get connectivity info from cloud."
-                                + " Check that the core device's IoT policy "
-                                + "grants the greengrass:GetConnectivityInfo permission");
-                return;
-            }
+            synchronized (shadowProcessingLock) {
```

Comment:
this isn't the technically best approach, but previous attempt, https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/76, ended up being complicated, so this may be the pragmatic approach.  open to suggestions of course